### PR TITLE
ingestion-beam style cleanup

### PR DESCRIPTION
**Pull request by jklukas**
A variety of changes to be more consistent in our style internally and more
consistent with some conventions of Beam.

We give all "user-facing" transforms private constructors and public static
factory methods, returning a Singleton instance in case of transforms that
take no parameters.

Also more consistent in skipping explicit names for transforms when they
simply repeat the class name.

### CODE REVIEW

File: ingestion-beam/src/test/java/com/mozilla/telemetry/decoder/DeduplicateTest.java [hunk 1]
```diff
@@ -39,8 +39,8 @@ public void testOutput() {
 
     // Create new PubsubMessage with element as document_id attribute
     final MapElements<String, PubsubMessage> mapStringsToId = MapElements
-        .into(new TypeDescriptor<PubsubMessage>() {
-        }).via(element -> new PubsubMessage(new byte[0], ImmutableMap.of("document_id", element)));
+        .into(TypeDescriptor.of(PubsubMessage.class))
+        .via(element -> new PubsubMessage(new byte[0], ImmutableMap.of("document_id", element)));
```
Comment (jklukas) [inline ingestion-beam/src/test/java/com/mozilla/telemetry/decoder/DeduplicateTest.java]: We remove a few instances of `new TypeDescriptor` with the simpler `TypeDescriptor.of`

File: ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/WithErrors.java [hunk 2]
```diff
@@ -85,7 +84,6 @@ public Pipeline getPipeline() {
     @Override
     public void finishSpecifyingOutput(String transformName, PInput input,
         PTransform<?, ?> transform) {
-      errors().setCoder(PubsubMessageWithAttributesCoder.of());
```
Comment (relud) [inline ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/WithErrors.java]: do we just not need this anymore?

Comment (jklukas) [reply ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/WithErrors.java]: I think this depends on what type information is available to the particular transform. There is a `PubsubCoderProviderRegistrar` that gets loaded as part of the google cloud beam package. In this particular case of the error collection for `WithErrors`, the collection has a concrete type of `PCollection<PubsubMessage>`, so it's able to use the registered coder.

For MapElementsWithErrors, the output collection type is a type parameter, so the coder registry is not able to infer. Thus, we still do need to explicitly set coder when the output collection type is pubsubmessage.

Comment (relud) [reply ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/WithErrors.java]: ah, makes sense. thanks for the explanation!

**Pull request merged**